editor/plugins/shader_editor_plugin.cpp

```diff
@@ -265,13 +265,19 @@ void ShaderEditorPlugin::get_window_layout(Ref<ConfigFile> p_layout) {
 	for (int i = 0; i < shader_tabs->get_tab_count(); i++) {
 		EditedShader edited_shader = edited_shaders[i];
 		if (edited_shader.shader_editor || edited_shader.visual_shader_editor) {
-			shaders.push_back(edited_shader.shader->get_path());
+			String shader_path;
+			if (!edited_shader.shader.is_null()) {
+				shader_path = edited_shader.shader->get_path();
+			} else {
+				shader_path = edited_shader.shader_inc->get_path();
```

Comment:
Does it need to check also for shader_inc to not be null?

Reply:
In general, the behavior in these shader editors seems to be that the shader is checked first. If that is null, assume that shader include is valid. If both are null for some reason the code will crash because something else is very wrong and should be fixed. Probably not the most robust way to handle things, but seems to work okay in practice.

For example here:

https://github.com/godotengine/godot/blob/2eec9a67d564b11326f44c5ef8b6b6f9aec251b9/editor/plugins/shader_editor_plugin.cpp#L49-L54

Or in here (and several other places):

https://github.com/godotengine/godot/blob/2eec9a67d564b11326f44c5ef8b6b6f9aec251b9/editor/plugins/text_shader_editor.cpp#L429-L434

Reply:
In that case you can add a DEV_ASSERT inside the ``else`` condition just before reading from ``shader_inc``. DEV_ASSERTs are compiled out in release builds so they are very handy for this type of rare error checking

Reply:
Done.